client/core/trade.go

```diff
@@ -746,6 +759,12 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 		}
 	}
 
+	// Check if we need to return the change coin because the order has been
+	// canceled.
+	if t.metaData.Status == order.OrderStatusCanceled && !t.hasUnswappedMatches() {
```

Comment:
A comment I made in my draft PR at https://github.com/decred/dcrdex/commit/98cc3469844e91966bd009841c43e30682e46a07#diff-5c4d3ee4527d6c50d1379b8edb5ef11eR221-R224 was about the locking around these `trackedTrade` fields.  Are we sure there's concurrent read/writes possible?  It's not clear to me what `trackedTrade.mtx` is meant to guard.